### ISSUE TITLE
clustype_noClus vs. clustype_inexact vs. clustype_exact

### ISSUE THREAD

**Issue by abisee** [shanzhenren/ClusType#3]
Hello,

I notice that `step1-entity_recognition.py` calls `clustype_noClus` but has commented out a call to`clustype_inexact`. Looking in `clustype.py` I see that there is also a `clustype_exact` function. These functions seem quite similar -- could you explain what is the difference between them?

Thanks!

**Comment by abisee** [comment]
Hi,

I've looked at the [paper](https://github.com/shanzhenren/ClusType/issues/3) in more detail and can see that a ClusType variant called ClusType-NoClus is described there, which presumably corresponds to `clustype_noClus` in the code.

My remaining questions are:
- What's the difference between the functions `clustype_inexact`, `clustype_exact`, `clustype_uniform`,  `clustype_noFsFc` provided in `clustype.py`?
- Which of these functions corresponds to the full version of ClusType reported in the [paper](https://github.com/shanzhenren/ClusType/issues/3)? (for example, the last row in Table 5)

Thanks!

**Comment by shanzhenren** [comment]
Hi @abisee ,

Sorry for the confusion. I have cleaned up the codes in algorithm.py and clustype.py to remove the unused functions (`clustype_exact`, `clustype_uniform`, `clustype_noFsFc`). 

 `clustype_inexact` (as put in `step1-entity_recognition.py`) corresponds to the final model reported in Table 5. It follows Algorithm 1 in the [paper](https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/kdd15clustype.pdf). `clustype_exact` are running a converged NMF for inner loop of the algorithm, which will take much more time but gives similar results. Other functions are just different variants for debug and I have removed them.

I have change `clustype_noClus` to `clustype_appx` as it works as a "warm start" for the final model.
